### PR TITLE
Release/0.7.6

### LINKED ISSUE #174: Set X-Username header for 429 responses
This header is currently only added for logged in users and OAuth requests. We want to expand this to include 429 responses for too many failed login attempts.

### LINKED ISSUE #175: Switch to production-hardened runtime (Gunicorn)
Switch the application runtime from Werkzeug test server to Gunicorn.

### LINKED ISSUE #179: Run Alembic DB migration/upgrade on start-up
- [x] Add `flask prod_run` Click command
- [x] Before invoking Werkzeug server, runs `flask db upgrade`

### PR DESCRIPTION
**Pull request by vtepliuk**
v. 0.7.6
--------

* Production hardening (#179, #175, #174)

(Resolves #179, resolves #175 and resolves #174.)

### CODE REVIEW

File: Jenkinsfile [hunk 1]
```diff
@@ -55,8 +59,10 @@ pipeline {
                             case 'release':
                                 assert branchTypeAndName[1] == projectVersion
                                 return "${projectVersion}-rc.${buildNumber}"
+                            case 'PR':
+                                return "${projectVersion}+PR.${branchTypeAndName[1]}.${buildNumber}"
                             default:
-                                throw new AssertionError("Oops, Mats messed up! :(")
+                                error "Oops, we messed up! :("
```
Comment (mxtthias) [inline Jenkinsfile]: :+1:

File: gunicorn_conf.py [hunk 2]
```diff
@@ -0,0 +1,32 @@
+# -*- coding: utf-8 -*-
+"""Gunicorn production config."""
+
+from __future__ import absolute_import, division, print_function, unicode_literals
+
+from click import echo
+
+bind = '0.0.0.0:5000'
+
+preload_app = True
```
Comment (mxtthias) [inline gunicorn_conf.py]: Maybe add a comment about why we want this? Same goes for `gthread` below. It's not obvious why these are our desired settings from their name alone.

File: messages.pot [hunk 3]
```diff
@@ -1,14 +1,14 @@
 # Translations template for xl_auth.
-# Copyright (C) 2017 National Library of Sweden
+# Copyright (C) 2018 National Library of Sweden
```
Comment (mxtthias) [inline messages.pot]: Autoupdate :(

Comment (mblomdahl) [reply messages.pot]: Yeah.. I'm thinking "let's keep it as this Babel tool wants it", else we'll have to go revert it manually on each `flask translate`. Seems like a waste of time..

**Pull request merged**